detector/PID/ARC_geo_v0.cpp

```diff
@@ -0,0 +1,871 @@
+//----------------------------------
+//         ARC detector v0
+//----------------------------------
+
+/*!
+ *  \brief     Detector constructor of barrel of ARC detector
+ *  \details   This code creates full geometry of ARC detector
+ *             Evolved from the pfRICH example in DD4hep.
+ *  \author    Alvaro Tolosa-Delgado alvaro.tolosa.delgado@cern.ch
+ *  \author    Martin Tat            martin.tat@cern.ch
+ *  \version   0
+ *  \date      2023
+ *  \pre       DD4hep compiled with Geant4+Qt
+ *  \bug       Walls do not reflect optical photons. Hard-coded values in many places.
+ */
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/OpticalSurfaces.h"
+#include "DD4hep/Printout.h"
+
+using namespace dd4hep;
+
+// #define DUMP_SENSOR_POSITIONS
+
+/// Function to build ARC endcaps
+static Ref_t create_arc_endcap_cell(Detector &desc, xml::Handle_t handle, SensitiveDetector sens)
+{
+  xml::DetElement detElem = handle;
+  std::string detName = detElem.nameStr();
+  int detID = detElem.id();
+//   xml::Component dims = detElem.dimensions();
+  OpticalSurfaceManager surfMgr = desc.surfaceManager();
+  DetElement det(detName, detID);
+  sens.setType("tracker");
+
+  /// z position of center of endcap. Original design 220cm
+  double zpos_endcap = detElem.attr<double>(_Unicode(zpos));
+
+  auto gasElem    = detElem.child(_Unicode(radiatorgas));
+  auto gasvolMat  = desc.material(gasElem.attr<std::string>(_Unicode(material)));
+  auto gasvolVis  = desc.visAttributes(gasElem.attr<std::string>(_Unicode(vis)));
+
+  auto vesselElem = detElem.child(_Unicode(vessel));
+  auto vesselMat  = desc.material(vesselElem.attr<std::string>(_Unicode(material)));
+  auto vesselVis  = desc.visAttributes(vesselElem.attr<std::string>(_Unicode(vis)));
+
+  // mother volume corresponds to the world
+  Volume motherVol = desc.pickMotherVolume(det);
+
+  // // // // // // // // // // // // // // // // // // // // // // // // // //
+  // // // // // // // //          VESSEL PARAMETERS          // // // // // //
+  // // // // // // // // // // // // // // // // // // // // // // // // // //
+  double vessel_outer_r = 210 * cm; // 190 * cm;
```

Comment:
are these parameter values going to configurable via XML?

Reply:
the position of sensors and mirrors is optimized for a given geometry (inner and outer radius mainly). I can move the definition of all parameters to the XML, but the user would have the responsibility of re-optimizing the position of sensors/mirrors again. Do you think it is better to keep the definition of such parameters in the XML?

Reply:
As long as it is  clearly stated that a re-optimization has to be performed (I would propose to write it both as a comment in the xml next to  the radius definition and in the README) I think it is better to put those values in the xml since we hope to get at some point a script automatizing this optimization for an arbitrary radius.